test/test_compile.py

```diff
@@ -70,6 +70,64 @@ def test_compile_with_simple_sketch(run_command, data_dir):
     assert is_message_sequence_in_json_log_traces(expected_trace_sequence, json_log_lines)
 
 
+def test_compile_with_sketch_with_symlink_selfloop(run_command, data_dir):
+    # Init the environment explicitly
+    result = run_command("core update-index")
+    assert result.ok
+
+    # # Download latest AVR
+    result = run_command("core install arduino:avr")
+    assert result.ok
+
+    sketch_name = "CompileIntegrationTestSymlinkSelfLoop"
+    sketch_path = os.path.join(data_dir, sketch_name)
+    fqbn = "arduino:avr:uno"
+
+    # Create a test sketch
+    result = run_command("sketch new {}".format(sketch_path))
+    assert result.ok
+    assert "Sketch created in: {}".format(sketch_path) in result.stdout
+
+    # create a symlink that loops on himself
+    loop_file_path = os.path.join(sketch_path, "loop")
+    os.symlink(loop_file_path, loop_file_path)
+
+    # Build sketch for arduino:avr:uno
+    result = run_command(
+        "compile -b {fqbn} {sketch_path}".format(
+            fqbn=fqbn, sketch_path=sketch_path))
+    # The assertion is a bit relaxed in this case because win behaves differently from macOs and linux
+    # returning a different error detailed message
+    assert "Error during sketch processing" \
+           "".format(
```

Comment:
is the `format` working? I don't see the replacement field

Reply:
see https://github.com/arduino/arduino-cli/pull/462/commits/3db1dc953b71f4415c4d3c5959c56254f0fdbf5d